Suggestion to optimize borrowing and change formatting.

elichai opened the pull request:
- Changed every ownership take that was unnecessary with borrowing.

- Changed every unnecessary cloning with ownership take (it's better to let the caller of a function decide if he still needs the variable and should clone or not and just give ownership).

- Replaced every `&Vec<T>` with `&[T]`, This way even if the caller doesn't have a vector he can still pass it to the function and if he does have a vector he can just use `[..]`

- Formatted some parts to be more rust idiomatic

elichai committed:
Replaced borrowing with consuming wherever the var was cloned

elichai committed:
Implemented TODO:: add create from private key

elichai committed:
Fixed the tests after the cloning/borrowship changes

elichai committed:
Replaced consuming with borrowing wherever it was unnecessary

elichai committed:
Moved hex into dev-dependencies since it's used only in tests

elichai committed:
Formatting Issues

omershlo commented:
@elichai this is a fantastic pr! I will adapt https://github.com/KZen-networks/kms-secp256k1 accordingly

elichai commented:
Thanks.
Two thing I forgot to mention:
1. I changed `update_key_pair()` from returning a new `KeyPair` to updating the `self` one.
It sounds to me like that will be the common use case of this function and if the caller need to keep the old one the same he can always `let new = old.clone().update_key_pair(fe);`

2. I implemented the `TODO` you had to create from private keys here: https://github.com/KZen-networks/multi-party-schnorr/pull/22/files#diff-be59b1c755c4e442b77f9ff24b57b701R76 I hope this is what you meant
